src/cosp.F90

```diff
@@ -1818,9 +1818,22 @@ SUBROUTINE COSP_INIT(Lisccp, Lmodis, Lmisr, Lcloudsat, Lcalipso, LgrLidar532, La
     if (Lmodis) call cosp_modis_init()
     if (Lmisr)  call cosp_misr_init()
     if (Lrttov) call cosp_rttov_init()
-    if (Lcloudsat) call cosp_cloudsat_init(cloudsat_radar_freq,cloudsat_k2,              &
+    if (Lcloudsat) then
+       call cosp_cloudsat_init(cloudsat_radar_freq,cloudsat_k2,                          &
          cloudsat_use_gas_abs,cloudsat_do_ray,R_UNDEF,N_HYDRO, surface_radar,            &
          rcfg,cloudsat_micro_scheme)
+    else
```

Comment:
@alejandrobodas Why do we need to allocate these if the radar simulator is not being used?

Reply:
Thanks @dustinswales. This is not strictly necessary. It is only needed if the individual allocatable variables are passed through an argument list somewhere rather than the instance of the type as a whole, which I believe only happens at lower level (i.e. if the simulator is called). I've deleted the change. I've also added the relevant deallocate statements that were missing from the relevant subroutine in the driver.